### PR TITLE
feat: add helper to compute buyAmount with token precision

### PR DESCRIPTION
**Pull request by Sekhmet**
### Summary

This PR adds helper (getOrderBuyAmount) that computes accurate buyAmount for order considering the given sellAmount and price.

This helper retains full precision of the token and ensures values are rounded up instead of truncated (for example price of 0.3 results in buyAmount of 3.333334 instead of 3.333333 (which makes actual price lower than expected instead of higher).

### How to test

1. Go there: http://localhost:8080/#/auction/sep:121
2. Submit some order and inspect computed values in your transaction (possible edge case to test is `1 SNUSDC` at `0.3 SNUSDC/TEST`.

### Screenshots

image

### CODE REVIEW

File: apps/ui/src/components/FormAuctionBid.vue [hunk 1]
```diff
@@ -213,29 +214,34 @@ const hasErrors = computed<boolean>(() => {
...
 });
 
-function toRawUnits(value: number, decimals: number): bigint {
-  return BigInt(Math.floor(value * 10 ** decimals));
+// TODO: decide what to do with this function.
+// We could write our own variant and use it throughout the codebase.
+// Or we could just use ethers.js parseUnits directly.
+// Plain parseFloat() * 10 ** decimals could lead to precision issues.
+function toRawUnits(value: string, decimals: number): bigint {
+  return parseUnits(value, decimals).toBigInt();
```
[the remark this answers is not in the document]
Comment (ChaituVR) [reply apps/ui/src/components/FormAuctionBid.vue]: Number 2 makes sense, atleast for now
Number 1 is not too bad, as long as AI can swap everything later
